runtime/lib/ttnn/operations/conv/conv2d.cpp

```diff
@@ -16,6 +16,9 @@ void run(const ::tt::target::ttnn::Conv2dOp *op, ProgramContext &context) {
       context.getDeviceFromView(op->device()->global_id(), 0);
   const ::ttnn::Tensor &input = tensorPool.at(op->input()->global_id());
   const ::ttnn::Tensor &weight = tensorPool.at(op->weight()->global_id());
+  assert(input.is_allocated());
```

Comment:
Why are we asserting this? Is this absolutely necessary? Just curious if you ever hit a case that this isn't true, since internally `is_allocated` calls `get_storage` which will be a performance hit once we implement multichip.

Reply:
I did when dealloc was forced. Do we have a debug only version of assert?

Reply:
That's a good point I'll add that in #908, I'll port your asserts to the debug-only version later, you can leave this as is for now. Otherwise LGTM!